src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -236,4 +227,26 @@ private static boolean checkIsAllLinesAreWhitespace(String[] lines, int lineFrom
         }
         return result;
     }
+
+    /**
+     * Calculates the left curly corresponding to the block to be checked.
+     *
+     * @param ast a {@code DetailAST} value
+     * @return the left curly corresponding to the block to be checked
+     */
+    private static DetailAST findLeftCurly(DetailAST ast) {
+        final DetailAST leftCurly;
+        if (ast.getType() == TokenTypes.LITERAL_CASE
+                && ast.getNextSibling() != null
+                && "{".equals(ast.getNextSibling().getFirstChild().getText())) {
```

Comment:
@romani Would it be alright if I redo the logic for `hasText` ? That's where the problem is I feel and I think @MEZk also feels that way ! 
Diff report: https://diff-report-issue3839.netlify.com/checkstyle/index.html